lib/collection/src/shard/replica_set.rs

```diff
@@ -22,7 +25,9 @@ use crate::operations::CollectionUpdateOperations;
 
 pub type IsActive = bool;
 pub type OnPeerFailure =
-    Box<dyn Fn(PeerId, ShardId) -> Box<dyn Future<Output = ()> + Send> + Send + Sync>;
+    Arc<dyn Fn(PeerId, ShardId) -> Box<dyn Future<Output = ()> + Send> + Send + Sync>;
+
+const READ_FAN_OUT_RATION: f32 = 0.33;
```

Comment:
I think I agree with this choice of magic number but what was the rational?

Reply:
I just remembered it from one of our discussions. Not sure if it is correct. Maybe @generall can check if this default makes sense.